waspc/src/Wasp/JsImport.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+
+module Wasp.JsImport
+  ( JsImport (..),
+    JsImportName (..),
+    JsImportData,
+    JsImportAlias,
+    JsImportPath,
+    getJsImportData,
+  )
+where
+
+import Data.Data (Data)
+import Data.List (isPrefixOf)
+import StrongPath (Dir', File', Path, Posix, Rel)
+import qualified StrongPath as SP
+
+data JsImport = JsImport
+  { -- | What is being imported.
+    _name :: JsImportName,
+    -- | Path from which we are importing.
+    _path :: JsImportPath,
+    -- | Alias for the imported name
+    _importAlias :: Maybe JsImportAlias
+  }
+  deriving (Show, Eq, Data)
+
+type JsImportPath = Path Posix (Rel Dir') File'
+
+type Identifier = String
+
+type JsImportAlias = String
+
+data JsImportName
+  = -- | Represents external imports like @import Identifier from "file.js"@
+    JsImportModule Identifier
+  | -- | Represents external imports like @import { Identifier } from "file.js"@
+    JsImportField Identifier
+  deriving (Show, Eq, Data)
+
+type JsImportIdentifier = String
+
+type JsImportWhat = String
+
+type JsImportStatement = String
+
+type JsImportData = (JsImportIdentifier, JsImportStatement)
+
+getJsImportData :: JsImportPath -> JsImportName -> Maybe JsImportAlias -> JsImportData
```

Comment:
Why don't we instead have function `getJsImportStmtAndIdentifier :: JsImport -> (JsImportIdentifier, JsImportStatement)`

Reply:
Btw do we have tests for this? If not, we should add some, this is a nice thing to test!

Reply:
We don't have tests atm, I'll add them 👍